### PR TITLE
Add sowbug/68keys and sowbug/ansi_tkl

### PR DESCRIPTION
**Pull request by sowbug**
## Description

Adds two new Blue Pill-based keyboards using reasonably common layouts (Magicforce 68 and ANSI tenkeyless), with
USB-C, optional Kailh hotswap sockets, per-key RGB lighting. Both keyboards have been built as prototypes.

These keyboards were generated with the help of https://github.com/sowbug/keycad.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

I read https://beta.docs.qmk.fm/developing-qmk/c-development/hardware_keyboard_guidelines and believe this PR complies.

### CODE REVIEW

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/sowbug/68keys/68keys.c [hunk 1]
```diff
@@ -0,0 +1,40 @@
+// 68keys.c
```
[the remark this answers is not in the document]
Comment (sowbug) [reply keyboards/sowbug/68keys/68keys.c:1]: done

File: keyboards/sowbug/68keys/68keys.c [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+#include "68keys.h"
+#include "quantum/rgb_matrix.h"
```
[the remark this answers is not in the document]
Comment (sowbug) [reply keyboards/sowbug/68keys/68keys.c:4]: done

File: keyboards/sowbug/68keys/68keys.c [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+led_config_t g_led_config = { {
```
[the remark this answers is not in the document]
Comment (sowbug) [reply keyboards/sowbug/68keys/68keys.c:6]: done

File: keyboards/sowbug/68keys/68keys.c [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+}
+};
```
[the remark this answers is not in the document]
Comment (sowbug) [reply keyboards/sowbug/68keys/68keys.c:20]: done

File: keyboards/sowbug/68keys/68keys.c [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+void matrix_init_kb(void) {
+	// put your keyboard start-up code here
+	// runs once when the firmware starts up
+
+	matrix_init_user();
+}
```
[the remark this answers is not in the document]
Comment (sowbug) [reply keyboards/sowbug/68keys/68keys.c:27]: done

File: keyboards/sowbug/68keys/config.h [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+#include "config_common.h"
+
+// USB Device descriptor parameter
+#define VENDOR_ID       0xFEED
+#define PRODUCT_ID      0x8888
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    Generic
+#define PRODUCT         Keyboard
+#define DESCRIPTION     Keyboard
```
[the remark this answers is not in the document]
Comment (sowbug) [reply keyboards/sowbug/68keys/config.h:13]: done

File: keyboards/sowbug/68keys/config.h [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+// in rules.mk.
+//
+// #define BACKLIGHT_PIN rgb_matrix
+#ifdef RGB_DI_PIN
+    #define RGB_MATRIX_KEYPRESSES
+    #define DRIVER_LED_TOTAL (68)
+    #define RGBLED_NUM (DRIVER_LED_TOTAL)
+#endif
+#define RGB_MATRIX_FRAMEBUFFER_EFFECTS
+#define RGB_MATRIX_MAXIMUM_BRIGHTNESS 128
```
[the remark this answers is not in the document]
Comment (sowbug) [reply keyboards/sowbug/68keys/config.h:40]: done

File: keyboards/sowbug/68keys/info.json [hunk 8]
```diff
@@ -0,0 +1,5 @@
+{
+    "keyboard_name": "68keys",
+    "url": "github.com/sowbug",
+    "maintainer": "sowbug"
```
[the remark this answers is not in the document]
Comment (sowbug) [reply keyboards/sowbug/68keys/info.json:4]: Oh, that's cool. My keyboard is based on Blue Pill, so not all the generated files applied, but the current info.json is now complete AFAICT.

File: keyboards/sowbug/68keys/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,31 @@
+// keymaps/default/keymap.c
```
[the remark this answers is not in the document]
Comment (sowbug) [reply keyboards/sowbug/68keys/keymaps/default/keymap.c:1]: done

File: keyboards/sowbug/68keys/68keys.h [hunk 10]
```diff
@@ -0,0 +1,16 @@
+// 68keys.h
```
[the remark this answers is not in the document]
Comment (sowbug) [reply keyboards/sowbug/68keys/68keys.h:1]: done

File: keyboards/sowbug/68keys/config.h (hunk 6 above)
[the remark this answers is not in the document]
Comment (sowbug) [reply keyboards/sowbug/68keys/config.h:13]: done (did all the corresponding changes to the new ansi_tkl keyboard as well)

File: keyboards/sowbug/68keys/readme.md [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+that treats RGB(0, 0, 0) as eligible for backlighting. Unfortunately the QMK
+maintainers weren't interested (actually, now that I look back, I didn't submit
+a pull request, but rather discussed it in a chat room), so it's not upstreamed.
+Thus you'll have to add that patch (and
+[this](https://github.com/sowbug/qmk_firmware/commit/ff17800ef4a9cba59d754c72720d6877edb8fbb9))
+to your copy of QMK and then enable it in config.h and rules.mk.
+
+## Building and flashing
+
+* `qmk compile -kb sowbug/68keys -km default`
```
[the remark this answers is not in the document]
Comment (sowbug) [reply keyboards/sowbug/68keys/readme.md:33]: done

File: keyboards/sowbug/68keys/68keys.c [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+
+// See readme for connection between backlight and RGB matrix.
+#ifdef BACKLIGHT_DRIVER
```
[the remark this answers is not in the document]
Comment (sowbug) [reply keyboards/sowbug/68keys/68keys.c:30]: done

File: keyboards/sowbug/68keys/config.h [hunk 13]
```diff
@@ -0,0 +1,40 @@
+// config.h
```
[the remark this answers is not in the document]
Comment (sowbug) [reply keyboards/sowbug/68keys/config.h:1]: done

File: keyboards/sowbug/68keys/info.json (hunk 8 above)
[the remark this answers is not in the document]
Comment (sowbug) [reply keyboards/sowbug/68keys/info.json:4]: done for both

**Comment by sowbug** [comment]
All comments address. Please re-review. Thanks!

**Review by drashna — APPROVED** [review]

**Comment by sowbug** [comment]
Thank you for the approval, @drashna. How do I get the upstream merge to happen? Is there anything else I need to do on my end?

**Review by spidey3 — APPROVED** [review]

**Comment by drashna** [comment]
Thanks

**Pull request merged**